zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -15,6 +15,131 @@ use crate::{
     transparent::{self, Script},
 };
 
+// Used by boilerplate code below.
+
+#[derive(Clone, Debug)]
+struct TransparentAuth<'a> {
+    all_prev_outputs: &'a [transparent::Output],
+}
+
+impl zcash_primitives::transaction::components::transparent::Authorization for TransparentAuth<'_> {
```

Comment:
Why don't we import `zcash_primitives::transaction` and other types so that the code is shorter?

Reply:
I think the reason is to explicitly express that `transaction` comes from `zcash_primitives`. We could then maybe use
```rust
use zcash_primitives::transaction as zp_tx;
```

Reply:
Exactly, since we're interfacing our types with `zcash_primitives` types, importing directly would be too confusing. Using the whole path makes it clear it's an external type. But your suggestion is great since we make it shorter but it's still clear, done in 944d4fd6d